File: leetcode.js/轮转数组.js
var rotate = function(nums, k) {
    const n = nums.length;
    if (n == 0) return nums; 
    k = k % n;
    
    nums.splice(0,0,...nums.splice(n-k,k));
    return nums
};
// 示例用法
const nums = [1, 2, 3, 4, 5, 6, 7];
const k = 3;
console.log('原数组:', nums);
const rotatedNums = rotate(nums, k);
console.log('轮转后的数组:', rotatedNums);

File: leetcode.py/轮转数组.py
class Solution(object):
    def rotate(self, nums, k):
        a = []
        n = len(nums)
        if n == 0:
            return nums
        k = k % n
        nums[:] = nums[n-k:] + nums[:n-k]
        return nums
# 以下是使用 Solution 类的 rotate 方法的示例
if __name__ == "__main__":
    solution = Solution()
    nums = [-1,-100,3,99]
    k = 2
    result = solution.rotate(nums, k)
    print(f"原始数组: {nums}")
    print(f"轮转 {k} 次后的数组: {result}")
